### ISSUE TITLE
Promote x86_64-linux-musl to tier 1 platform

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#8732]
`x86_64-linux-musl` is currently a [tier 2 platform](https://github.com/crystal-lang/crystal/wiki/Platform-Support#tier-2). With #7420 merged, automatic testing is now set up for this platform. Since Alpine Linux is already used in the compiler build workflow, it is naturally a core platform for Crystal.

Binary releases have already been provided for a long time in form of the generic Linux tarball.
That works for now, but a dedicated binary release which includes static libraries linked against musl-libc would be preferrable (see https://github.com/crystal-lang/distribution-scripts/pull/29#discussion_r364259425). We might want to implement that first before moving to tier 1. But apart from that, I don't think there's anything standing in the way of promotion.

**Comment by RX14** [comment]
Don't we still have some hacks around exceptions? Or is the shards `require "llvm"` not neccesary any more?

**Comment by j8r** [comment]
@RX14 That's not really linked to the platform itself, that's static linking, which appears to be only supported on Alpine: https://github.com/crystal-lang/crystal/issues/4276
On glibc based distribution static linking is not supported at all.

**Comment by stakach** [comment]
this dedicated binary release would be amazing

**Comment by beta-ziliani** [comment]
After [revamping the platform support page](https://crystal-lang.org/reference/master/syntax_and_semantics/platform_support.html), this issue came forward: we seem to have everything ticked to make x86_64-linux-musl a Tier 1 platform, except that, quoting the description of this issue, "a dedicated binary release which includes static libraries linked against musl-libc would be preferrable".

I might be wrong, but I consider this a "nice to have" but not really a blocking aspect.  We are providing binaries that successfully run tests, so I consider this platform to be in Tier 1.

If that's agreed, we can proceed with https://github.com/crystal-lang/crystal-book/pull/679

**Comment by stakach** [comment]
it would be nice to be able to install crystal via the alpine package manager - not sure if this is an official source?
https://pkgs.alpinelinux.org/packages?name=crystal&branch=edge&repo=&arch=&maintainer=

but that should be a sperate issue and otherwise agree with you and have not had any issues using crystal on musl and all our production releases run on it.

**Comment by jirutka** [comment]
Yes, we do package crystal for x86_64 and aarch64 in Alpine Linux, since Alpine v3.8 (2020). I’m the maintainer of this package. https://pkgs.alpinelinux.org/packages?name=crystal is the official source.

**Comment by stakach** [comment]
sweet! good to know @jirutka 
seems like there is nothing holding it back from Teir 1 then as the binaries are available via the alpine package manager
